exonum/src/node/state.rs

```diff
@@ -863,6 +920,38 @@ impl State {
         }
     }
 
+    /// Finds unknown transactions in the block.
+    pub fn unknown_block_txs(
+        &mut self,
+        msg: &BlockResponse,
+        txs: &MapIndex<&&Snapshot, Hash, RawMessage>,
+        txs_pool: &KeySetIndex<&&Snapshot, Hash>,
+    ) -> Result<&IncompleteBlock, failure::Error> {
+        match self.incomplete_blocks.entry(msg.block().hash()) {
+            Entry::Occupied(..) => bail!("Block already found"),
+            Entry::Vacant(e) => {
+                let mut unknown_txs = HashSet::new();
+                for hash in msg.transactions() {
+                    if txs.get(hash).is_some() {
+                        if !txs_pool.contains(hash) {
```

Comment:
I don't quite understand the reasoning to use `bail` here. If the `BlockResponse` is authenticated properly (which seems to be the case when the method is called from `handle_block`), then this should be a fundamental error.